agent/fw_wordpress.c

```diff
@@ -215,6 +215,26 @@ static void free_wordpress_metadata(void* metadata) {
   nr_free(metadata);
 }
 
+static inline void nr_wordpress_hooks_create_metric(nr_segment_t* segment,
+                                       const char* hook_name) {
+  if (NULL == segment) {
+    return;
+  }
+  // need to capture 'now' because segment has not finished yet and therefore
+  // can't use segment->stop_time
+  nrtime_t now = nr_txn_now_rel(segment->txn);
```

Comment:
Probably unlikely - but I think if `segment->txn` was NULL then `now` would be set to `0` and `duration` would always be greater than the `wordpress_hooks_threshold` and so this would create a whole lot of metrics.  Maybe a check for if `NULL == segment->txn` is needed?

Reply:
I don't think the `NULL == segment->txn` check is necessary here. `nr_time_duration(start, stop)` returns 0 if `start` (`segment->start_time`) is greater than `stop` (`now`). So if `now` is 0, because `segment->txn` is `NULL`, `duration` will be set to 0 and, unless `NRINI(wordpress_hooks_threshold)` is 0, no metric will be generated.